### ISSUE TITLE
Inherited socket is not used in AsyncSocket class

### ISSUE THREAD

**Issue by EnginEken** [EnginEken/tcppinglib#7]
A socket created in `async_tcpping()` function and send to AsyncSocket class but never used. Instead, new socket is created with asyncio loop's `create_connection()` function.

Calling `TcpSocket` class' `connect()` function in `AsyncSocket` class' `connect()` function would be more suitable since socket is already created and connect operation is already implemented.

It could be changed from this:
```python
if not self._tcp_sock:
            raise NotImplementedError

        loop = asyncio.get_running_loop()

        try:
            sock_port = int(request.port)

            self._tcp_sock._set_timeout(request.timeout)
            start_time = time.perf_counter()

            await loop.create_connection(
                asyncio.Protocol, host=request.destination, port=sock_port
            )

            request._time = time.perf_counter() - start_time

        except OSError as err:
            if err.errno == 8:
                raise NameLookupError(request.destination)

            raise TCPSocketError(str(err))
```

To this:

```python
if not self._tcp_sock:
            raise NotImplementedError

        try:
            self._tcp_sock._set_timeout(request.timeout)
            start_time = time.perf_counter()
            
            self._tcp_sock.connect(request)

            request._time = time.perf_counter() - start_time

        except OSError as err:
            if err.errno == 8:
                raise NameLookupError(request.destination)

            raise TCPSocketError(str(err))
```

**Comment by harrandt** [comment]
If you use ports that are not open in https://github.com/EnginEken/tcppinglib/blob/main/examples/async_multi_tcpping.py you will notice that the code is not async. Each request will hit the timeout 

You set the socket to `blocking=False` but since the `blocking` attribute is [only a wrapper for the timeout](https://docs.python.org/3/library/socket.html#socket.socket.setblocking) you are overwriting it two lines later.

https://github.com/EnginEken/tcppinglib/blob/main/tcppinglib/tcp_sockets.py#L238

To actually make it async you could use `await loop.create_connection` or `await asyncio.open_connection` [as shown here](https://github.com/anhenghuang/AsyncTCPPing/blob/master/__init__.py#L11).

This will actually wait for all timeouts at the same time and finishes after 1.5 secs instead of 1.5 * number of hosts (for count=1).

**Comment by EnginEken** [comment]
Thanks for your glorious comment @harrandt. I've totally missed that. To set the timeout properly, I also need to edit other TCP Socket classes' _set_timeout method. I'm working on it right now. I'll publish a new release when I did that.

**Comment by EnginEken** [comment]
This is fixed with #16. Closing the issue
